.eslintrc.cjs

```diff
@@ -0,0 +1,18 @@
+/* eslint-env node */
+module.exports = {
+  extends: [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended",
+    "prettier"
+  ],
+  parser: "@typescript-eslint/parser",
+  plugins: ["@typescript-eslint"],
+  root: true,
+  rules: {
+    "@typescript-eslint/no-unnecessary-type-constraint": "warn",
+    // Svelte doesn't correctly compile if imports of the generated /modules
+    // aren't imported as 'import type' in other parts of the generated
+    // querybuilder, so set this option to ensure we always do that
+    "@typescript-eslint/consistent-type-imports": "error"
```

Comment:
This ESLint rule replaces the now deprecated `tsconfig` flag `"importsNotUsedAsValues": "error"`. See [the PR introducing the change to TypeScript](https://github.com/microsoft/TypeScript/pull/52203) and [the section in the release announcement](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#verbatimmodulesyntax).

TL;DR: The intention of the `importsNotUsedAsValues` flag was never really to be a linting rule to disallow importing a type without the `import type` syntax. It also has weird interactions with `isolatedModules` and `preserveValueImports` which had some drastic effects on the transpiled JavaScript.